src/server/pfs/server/driver_new_storage.go

```diff
@@ -339,23 +339,28 @@ func deserializeShard(shardAny *types.Any) (*pfs.Shard, error) {
 	return shard, nil
 }
 
-// (bryce) it might potentially make sense to exit if an error occurs in this function
-// because each pachd instance that errors here will lose its compaction worker without an obvious
-// notification for the user (outside of the log message).
-// (bryce) ^ maybe just a retry would be good enough.
 func (d *driver) compactionWorker() {
 	w := work.NewWorker(d.etcdClient, d.prefix, storageTaskNamespace)
-	if err := w.Run(context.Background(), func(ctx context.Context, subtask *work.Task) error {
-		shard, err := deserializeShard(subtask.Data)
+	backoffStrat := backoff.NewExponentialBackOff()
+	backoffStrat.MaxElapsedTime = 0 // don't give up, compactionWorker should not exit
+
+	err := backoff.Retry(func() error {
```

Comment:
We should probably use the `RetryNotify` function and have the logging be done in the notify callback. This is the pattern we follow in the codebase because it is a bit cleaner and it more explicitly separates the logging of the function from the actual function (this would allow you to just return the results of the run function call in the operation callback).